src/main/groovy/com/readytalk/gradle/js/JsPlugin.groovy

```diff
@@ -22,40 +24,79 @@ class JsPlugin implements Plugin<Project> {
     this.project = project
 
     project.plugins.apply(BasePlugin)
-    project.plugins.apply(GruntPlugin)
-    project.plugins.apply(GulpPlugin)
+    project.plugins.apply(NodePlugin)
 
     if(!project.hasProperty('generateNodeWrapper')) {
       project.extensions.extraProperties.set('generateNodeWrapper', true)
     }
-    injectNodewSetup()
 
-    project.tasks.withType(GruntTask) {
-      dependsOn NpmInstallTask.NAME, GruntPlugin.GRUNT_INSTALL_NAME
+    nodeExt.with {
+      version = DEFAULT_NODE_VERSION
+      npmVersion = DEFAULT_NPM_VERSION
+      download = true
+      workDir = project.file("${project.buildDir}/nodejs")
     }
 
-    project.tasks.withType(GruntInstallTask) {
-      dependsOn NpmInstallTask.NAME
-    }
+    injectNodewSetup()
 
-    project.tasks.withType(GulpTask) {
-      dependsOn NpmInstallTask.NAME, GulpPlugin.GULP_INSTALL_NAME
+    //TODO: We ought to read the package.json for references to supported node versions
+    //      and default to those if not specified in the gradle extension
+    if (project.version == 'unspecified' && project.file('package.json').exists()) {
+      project.version = getPackageJsonVersion(project.file('package.json'))
     }
 
-    project.tasks.withType(GulpInstallTask) {
-      dependsOn NpmInstallTask.NAME
-    }
+    injectBinPath()
+  }
 
-    project.node {
-      version = '0.12.4'
-      npmVersion = '2.11.2'
-      download = true
-      workDir = project.file("${project.buildDir}/nodejs")
-    }
+  private NodeExtension getNodeExt() {
+    project.extensions.findByType(NodeExtension)
+  }
 
-    if (project.version == 'unspecified' && project.file('package.json').exists()) {
-      project.version = getPackageJsonVersion(project)
+  String getNodeHome() {
+    "${nodeExt.workDir}/node-v${nodeExt.version}-${new PlatformHelper().osName}-x64"
+  }
+
+  def injectBinPath() {
+    project.tasks.withType(NodeTask).all { NodeTask task ->
+      //Inject node_modules/.bin directory into path
+      //TODO: This really belongs upstream, since it hijacks the execOverrides block
+      //      I can't preserve the closure because the upstream plugin inexplicably marks it write-only
+      task.setExecOverrides {
+        it.environment.PATH =
+                "${nodeHome}/bin:${nodeExt.nodeModulesDir}/node_modules/.bin:${System.env.PATH}"
```

Comment:
It will now - I added a dummy run of the `nodew` script when it gets generated, both to check that it's operational and print a warning if not, and to ensure the symlinks are immediately generated for `node` and `npm` in the `.bin` directory.